src/compiler/scala/tools/nsc/backend/jvm/BTypesFromSymbols.scala

```diff
@@ -96,17 +96,15 @@ abstract class BTypesFromSymbols[G <: Global](val global: G) extends BTypes {
       cachedClassBType(internalName) match {
         case Some(bType) =>
           if (currentRun.compiles(classSym))
-            assert(classBTypeCacheFromSymbol.contains(internalName), s"ClassBType for class being compiled was already created from a classfile: ${classSym.fullName}")
+            assert(bType fromSymbol, s"ClassBType for class being compiled was already created from a classfile: ${classSym.fullName}")
           bType
         case None =>
-          // The new ClassBType is added to the map in its constructor, before we set its info. This
+          // The new ClassBType is added to the map via its apply, before we set its info. This
           // allows initializing cyclic dependencies, see the comment on variable ClassBType._info.
-          val res = ClassBType(internalName)(classBTypeCacheFromSymbol)
-          if (completeSilentlyAndCheckErroneous(classSym)) {
-            res.info = Left(NoClassBTypeInfoClassSymbolInfoFailedSI9111(classSym.fullName))
-            res
-          } else {
-            setClassInfo(classSym, res)
+          ClassBType(internalName, true) { res:ClassBType =>
+            if (completeSilentlyAndCheckErroneous(classSym))
```

Comment:
I would assume that we need to take `frontEndLock` here, as this calls into `Symbol.info`.

Reply:
As I understand it if we have access to the Symbol then we have the lock, so the caller already has the lock, We take the lock when access via some API that uses Symbol or other global data indirectly